src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala

```diff
@@ -122,7 +122,11 @@ class FeelInterpreter {
       case In(x, test) =>
         withVal(eval(x), x => eval(test)(context + (inputKey -> x)))
       case InstanceOf(x, typeName) =>
-        withVal(eval(x), x => withType(x, t => ValBoolean(t == typeName)))
+        typeName match {
+          case "Any" if eval(x) != ValNull => ValBoolean(true)
+          case _ =>
+            withVal(eval(x), x => withType(x, t => ValBoolean(t == typeName)))
+        }
```

Comment:
We can optimize the expression a bit to evaluate `x` only once. 

```suggestion
        withValOrNull(eval(x), x => {
          typeName match {
            case "Any" if x != ValNull => ValBoolean(true)
            case _ => withType(x, t => ValBoolean(t == typeName))
          }
        })
```

Reply:
Beautiful 😄